### PR TITLE
Fixes #19746: Move SELinux policy application into postinst script

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/19746

### CODE REVIEW

File: relay/sources/rudder-server-relay-postinst [hunk 1]
```diff
@@ -105,6 +106,33 @@ fi
...
+    # their context to the contexts defined in SELinux configuration,
+    # including the file contexts defined in the rudder-relay module
+    restorecon -R /var/rudder/inventories
+    restorecon -R /var/rudder/reports
+    restorecon -R /var/log/rudder/apache2
+    restorecon -R /opt/rudder/etc/relayd
+    restorecon /opt/rudder/bin/rudder-relayd
+    restorecon -R /var/rudder/lib
+    restorecon -R /var/rudder/cfengine-community/ppkeys
+    restorecon -R /var/rudder/share
```
Comment (ncharles) [inline relay/sources/rudder-server-relay-postinst:126]: is there a way to detect that this changed ? restorecon on /var/rudder/share can be painfully slow

File: relay/sources/rudder-server-relay-postinst [hunk 2]
```diff
@@ -105,6 +106,33 @@ fi
...
+if [ "${SELINUX}" -eq 1 ]; then
+  # Check "sestatus" presence, if not there there is not SELinux
+  # setup on this system
+  if type sestatus >/dev/null 2>&1 && sestatus | grep -q "enabled"; then
+    # Add/Update the rudder-relay SELinux policy
+    semodule -i /opt/rudder/share/selinux/rudder-relay.pp
+    # Ensure inventory directories context is set by resetting
+    # their context to the contexts defined in SELinux configuration,
+    # including the file contexts defined in the rudder-relay module
+    restorecon -R /var/rudder/inventories
```
Comment (ncharles) [inline relay/sources/rudder-server-relay-postinst:119]: is there a way to detect that this changed ? restorecon on /var/rudder/inventories can be painfully slow

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: relay/sources/rudder-server-relay-postinst [hunk 3]
```diff
@@ -105,6 +106,43 @@ fi
...
+  # setup on this system
+  if type sestatus >/dev/null 2>&1 && sestatus | grep -q "enabled"; then
+    # Add/Update the rudder-relay SELinux policy
+    semodule -i /opt/rudder/share/selinux/rudder-relay.pp
+    # Ensure inventory directories context is set by resetting
+    # their context to the contexts defined in SELinux configuration,
+    # including the file contexts defined in the rudder-relay module
+    #
+    # Test current context for big folders
+    if ! ls -Z /var/rudder/inventories/ | head -n1 | grep -q public_content_rw_t; then
```
Comment (ncharles) [inline relay/sources/rudder-server-relay-postinst:121]: <3

**Review by ncharles — APPROVED** [review]
:heart:

**Pull request merged**